File: Employee_System_Backend/src/main/java/com/project/service/EmpService.java
package com.project.service;

import java.util.List;

import com.project.model.Emp;

public interface EmpService {
	
	public Emp createEmp(Emp emp);
	
	public List<Emp> getAllEmp();
	
	public Emp getEmptyById(int id);
	
	public void deleteEmp(int id);
	
	public Emp updateEmp(int id, Emp emp);
}

File: Employee_System_Frontend/src/service/empservice.js
import axios from "axios";

const BASE_API_URL = "https://employee-data-management-system-dpq8.onrender.com/api/v1";

class EmpService {

  saveEmp(emp) {
    return axios.post(`${BASE_API_URL}/save`, emp);
  }

  getAllEmp() {
    return axios.get(`${BASE_API_URL}/`);
  }

  getEmpById(id) {
    return axios.get(`${BASE_API_URL}/${id}`);
  }

  deleteEmp(id) {
    return axios.delete(`${BASE_API_URL}/delete/${id}`);
  }

  updateEmp(id, emp) {
    return axios.put(`${BASE_API_URL}/update/${id}`, emp);
  }
}

const empService = new EmpService();
export default empService;
